### ISSUE TITLE
Serialized preFetch calls

### ISSUE THREAD

**Issue by Blaconix** [quasarframework/quasar-cli#173]
### Software version

OS: Windows 10 Pro
Node: v10.0.0
NPM: v6.0.1
Any other software related to your bug: 
    Quasar CLI........ v0.17.13
    Quasar Framework.. v0.17.10

### What did you get as the error?
* In my Default.vue Layout I'm fetching the user from the API endpoint using axios
```
const token = localStorage.getItem('token')
axios.defaults.headers.common['Authorization'] = `Bearer ${token}`

return axios.get('users/me')
  .then(response => {
    store.state.user = response.data.data
    store.state.token = token
  })
  .catch(() => {
    localStorage.removeItem('ety-token')
    redirect('/auth/login')
  })
```

This populates the store state with the user object.
* Now, on a child route, a new page (from Default Layout), when trying to use preFetch like so:
```
preFetch ({ store, redirect }) {
  if (store.state.user.group !== 'Administrator') {
    return redirect('/')
  }
}
```
**The` store.state.user` is not defined, `store.state` being only an Observe object.**
**This behavior only happens when accessing the route directly, by either entering the URL or refreshing the app on that route.** 
If accessed through another route, it works as expected.
I also tried enclosing the axios call in a separate Promise, which resulted the same.

I also tried with a separate store, as described here https://quasar-framework.org/guide/app-prefetch-feature.html#Store-Code-Splitting

I replaced the axios call with a `store.dispatch` which basically calls the axios then commits a mutation to fill the store. Same thing happens, store is not initialized correctly when accessing the child page directly.

### What were you expecting?
Store state to be initialized in `preFetch()` of children routes when route is accessed directly (through URL or refreshing the app)

**Comment by rstoenescu** [comment]
Hi,

You’re doing async work to fill in the store. So use preFetch (in App.vue, for example) and return the Promise of your Ajax call. Otherwise, your app gets initialized and launched then your async work kicks in… which is too late.

**Comment by Blaconix** [comment]
Here’s a repo showcasing the bug: https://github.com/Blaconix/quasar-prefetch-bug

Practically the children preFetch’es don’t wait for the parent ones to finish (and the page doesn’t render).

**Comment by rstoenescu** [comment]
Thank for the detailed explanations. Added serialized preFetch calls. Will be available in next Quasar CLI v0.17.14.
